sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSource.cs

```diff
@@ -111,7 +111,15 @@ internal DataFeedDetail InstantiateDataFeedDetail(string name, DataFeedGranulari
                 { DataSourceParameter = new() { Query = d.Query, ConnectionString = d.ConnectionString } },
             SqlServerDataFeedSource d => new SQLServerDataFeedPatch()
                 { DataSourceParameter = new() { Query = d.Query, ConnectionString = d.ConnectionString } },
-            _ => throw new InvalidOperationException("Invalid DataFeedDetailPatch type")
+            _ => new DataFeedDetailPatch() { DataSourceType = DataSourceKind }
```

Comment:
This is the "public model" => "internal update model" conversion. Also, no need to create a new "Unknown" type for this one, since it's not abstract.